src/elements/Input/Input.js

```diff
@@ -88,7 +88,13 @@ function Input(props) {
   )
   const rest = getUnhandledProps(Input, props)
   const ElementType = getElementType(Input, props)
-  const inputProps = _.pick(props, htmlInputPropNames)
+  const inputProps = {
+    ..._.pick(props, htmlInputPropNames),
+    // React gives a warning if no onChange is given to a controlled input. We
+    // handle the onChange on the wrapping element, so we don't actually need
+    // onChange on the input itself but let's pass a noop to suppress the warning.
+    onChange: _.noop,
```

Comment:
Actually, this causes the `handles events transparently` test to fail. I think that test tries to call the event (`onChange` in this case) on the top-level component (the `Input` in this case). Since the `onChange` is now actually being passed as a prop to the `htmlInput`, it's not being called when simulated on the `Input`.

Thoughts on how to handle this? I agree it would be better to not suppress the warning, so ideally we can improve the test, assuming I'm understanding the issue correctly.